include/aspect/postprocess/matrix_statistics.h

```diff
@@ -0,0 +1,57 @@
+/*
+  Copyright (C) 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__postprocess_matrix_statistics_h
+#define __aspect__postprocess_matrix_statistics_h
+
+#include <aspect/postprocess/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+
+    /**
+     * A postprocessor that computes some statistics about the matrices
+     *
+     * @ingroup Postprocessing
+     */
+    template <int dim>
+    class MatrixStatistics : public Interface<dim>, public ::aspect::SimulatorAccess<dim>
+    {
+      private:
+        std::string
+        get_stats(const TrilinosWrappers::BlockSparseMatrix &matrix, std::string matrix_name);
```

Comment:
I did apparently not look closely enough to this patch. In this function, can you:
* make the second argument a const reference
* break arguments onto separate lines
* I suspect that the function could be made `const` itself. In fact, I think the function doesn't actually have to be a member function, in which case it could just as well live in an anonymous namespace in the `.cc` file, rather than being part of the public interface.